packages/driver/src/cy/multi-domain/validator.ts

```diff
@@ -21,12 +21,19 @@ export class Validator {
       })
     }
 
-    if (data && !Array.isArray(data)) {
+    if (options && (!isObject(options) || Array.isArray(options))) {
       this.onFailure()
 
-      $errUtils.throwErrByPath('switchToDomain.invalid_data_argument', {
+      $errUtils.throwErrByPath('switchToDomain.invalid_options_argument', {
         onFail: this.log,
-        args: { arg: $utils.stringify(data) },
+        args: { arg: $utils.stringify(options) },
+      })
+    } else if (options && !Object.getOwnPropertyDescriptor(options, 'args')) {
+      this.onFailure()
+
+      $errUtils.throwErrByPath('switchToDomain.incomplete_options_argument', {
+        onFail: this.log,
+        args: { arg: $utils.stringify(options) },
```

Comment:
I don't think we should throw in this case. It's fine for there to be an empty options object and in the future, it will be fine for it only have other options and not the `args` property.

Reply:
Could we throw if they provide any keys other than the ones we allow? We'd only allow args for now, but that would cover the case where they are a passing in the wrong args at the wrong level.

Reply:
I like @mjhenkes's idea.

Reply:
@mjhenkes I'll update the PR to favor checking for keys other than that which we allow. I'll probably opt for something similar to how we verify the cypress config

Reply:
updated in 2f8317fc11b96d16a5c770124ac4c249572f5eb5